### ISSUE TITLE
ScrollView with BindableLayout.ItemsSource is not recalculate new size on iOS

### ISSUE THREAD

**Issue by KSemenenko** [dotnet/maui#12727]
### Description

I have this code, and after I changed Slots collection, ScrollView is not recalulate it's size.

``` xaml
<ScrollView x:Name="scrollView" VerticalScrollBarVisibility="Default">
<Grid>
<StackLayout Margin="16,0">
<StackLayout x:Name="DaysCollection" BindableLayout.ItemsSource="{Binding Days}" Spacing="18">
	<BindableLayout.ItemTemplate>
		<DataTemplate x:DataType="general:AvailabilityDayModel">
			<StackLayout Spacing="20">
				<StackLayout x:Name="AvailabilitySlot" BindableLayout.ItemsSource="{Binding Slots}" Spacing="4">
					<BindableLayout.ItemTemplate>
						<DataTemplate x:DataType="general:AvailabilitySlotModel">
							<HorizontalStackLayout>
								<Label Text="-" Style="{StaticResource ParagraphRegular}" Margin="8" VerticalOptions="Center" />
								<Button ImageSource="trash.png" Clicked="RemoveAvailability_Clicked" Margin="-8"
														  CommandParameter="{Binding .}" BackgroundColor="Transparent" VerticalOptions="Center"
														  Style="{StaticResource SecondaryDefaultSmall}"   />

								<Button ImageSource="plus_big.png"  Clicked="AddAvailability_Clicked" Margin="-8"
									CommandParameter="{Binding ParentDay}"  BackgroundColor="Transparent" VerticalOptions="Center"
									Style="{StaticResource SecondaryDefaultSmall}"  />
							</HorizontalStackLayout>
						</DataTemplate>
					</BindableLayout.ItemTemplate>
				</StackLayout>

			</StackLayout>
		</DataTemplate>
	</BindableLayout.ItemTemplate>
</StackLayout>

<Button IsVisible="{OnIdiom false, Phone=true}" Text="Save" Command="{Binding SaveChangesCommand}" Margin="16,32,16,0" Style="{StaticResource MainDefaultBig}" />
</StackLayout>
</Grid>
</ScrollView>
```

### Steps to Reproduce

1. Create ScrollView
2. Inside ScrollView add  Grid, then StackView
3. Add one more StackView with BindableLayout.ItemsSource
4. add one more butto wich will add new item into BindableLayout.ItemsSource and will rise event to add one more item
5. on iOS after adding items ScrollView will not recalulate size and you will not see scroll
6. on Android all is ok

### Link to public reproduction project repository

no

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16

### Did you find any workaround?

If I will call InvalidateMeasure(); after adding items in Page all is fine, so it looks like ScrollView not notified about chagin size of childrens

**Comment by KSemenenko** [comment]
[BindableLayoutIssue.zip](https://github.com/dotnet/maui/files/10471921/BindableLayoutIssue.zip)

I can't reproduce my problem specifically, maybe because I now have a dotnet 7.200 preview
But in my example there is a new problem, that after adding an internal element, the application no longer responds to buttons.

**Comment by KSemenenko** [comment]
ah, need a repo

**Comment by Sokol2001** [comment]
https://github.com/Sokol2001/BindableLayoutIssue

**Comment by KieranMaclagan** [comment]
Switching to a CollectionView is the only workaround I've found so far. Here's another repo showing ScrollView not working, and CollectionView working https://github.com/KieranMaclagan/ScrollViewTest.App

**Comment by KSemenenko** [comment]
Is CollectionView ready for use?

**Comment by matt-goldman** [comment]
Issue also occurs on Windows. I think this needs some urgent attention. Looks like there are related issues too (#132466 and #11987).

**Comment by planerde** [comment]
I have a similar issue, but the workaround mentioned in https://github.com/dotnet/maui/issues/9825#issuecomment-1346691949 is working for me

**Comment by noque-lind** [comment]
> [BindableLayoutIssue.zip](https://github.com/dotnet/maui/files/10471921/BindableLayoutIssue.zip)
> 
> I can't reproduce my problem specifically, maybe because I now have a dotnet 7.200 preview But in my example there is a new problem, that after adding an internal element, the application no longer responds to buttons.

Have the exact same issue!

**Comment by MichaelNovikov** [comment]
> [BindableLayoutIssue.zip](https://github.com/dotnet/maui/files/10471921/BindableLayoutIssue.zip)
> 
> Конкретно свою проблему воспроизвести не могу, может потому что у меня сейчас dotnet 7.200 preview Но в моем примере появилась новая проблема, что после добавления внутреннего элемента приложение перестало реагировать на кнопки.

Similarly, ios stopped responding to clicks on elements.

**Comment by hugebug4ever** [comment]
exactly the same issue

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 2.0. Can repro on iOS platform with sample project.
[BindableLayoutIssue.zip](https://github.com/dotnet/maui/files/11933556/BindableLayoutIssue.zip)
![12727](https://github.com/dotnet/maui/assets/77434865/d4031a95-ffe0-4686-b38d-705eee73be54)

**Comment by MateuszLas421** [comment]
the same issue

**Comment by i4GAC** [comment]
Exactly the same problem, workaround with CollectionView not possible due to requirements. I should add that this happens when the DataTemplate doesn't fit on the screen and part of it is hidden. After scrolling, all the buttons below StackLayout  don't work. If the buttons are above the StackLayout then the buttons work fine.

**Comment by hartez** [comment]
It's possible this is fixed by #16385.

**Comment by hartez** [comment]
> It's possible this is fixed by #16385.

Confirmed, fixed by #16385.
